.github/workflows/build.yaml

```diff
@@ -0,0 +1,96 @@
+on:
+  workflow_call:
+    outputs:
+      artefact:
+        description: The name of the uploaded image aretfact.
+        value: ${{ jobs.build.outputs.artefact }}
+      tag:
+        description: The tag used for the built image.
+        value: ${{ jobs.build.outputs.tag }}
+
+jobs:
+  build:
+    name: Build snekbox-venv image
+    runs-on: ubuntu-latest
+    outputs:
+      artefact: ${{ env.artefact }}
+      tag: ${{ steps.sha_tag.outputs.tag }}
+    env:
+      artefact: image_artefact_snekbox-venv
+
+    steps:
+      # Create a short SHA with which to tag built images.
+      - name: Create SHA Container Tag
+        id: sha_tag
+        run: |
+          tag=$(cut -c 1-7 <<< $GITHUB_SHA)
+          echo "::set-output name=tag::$tag"
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # The current version (v2) of Docker's build-push action uses buildx,
+      # which comes with BuildKit. It has cache features which can speed up
+      # the builds. See https://github.com/docker/build-push-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN  }}
+
+      # The image built for PRs may start to deviate from the "latest" image
+      # currently in GHCR. Configure the subsequent build step to cache the
+      # layers in GitHub Actions for PRs.
+      # See https://github.com/moby/buildkit#github-actions-cache-experimental
+      #
+      # Because the cache is scoped to the branch, it will not be available
+      # on the main branch when the PR is merged. Furthermore, using this cache
+      # on main is redundant since the previous build's images are already
+      # cached on GHCR. Thus, this step is only used for PRs.
+      - name: Configure cache
+        id: cache_config
+        run: |
+          set -euo pipefail
```

Comment:
this line is interesting, especially `pipefail`. What does it do and why is it relevant? I don't know if we need to document it better or not, maybe not.

Reply:
The pipefail is actually not relevant here since there aren't any pipes. I could remove it to reduce confusion. But in general these options are really nice https://coderwall.com/p/fkfaqq/safer-bash-scripts-with-set-euxo-pipefail